Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/conceptcollection/ConceptCollectionDAO.java

```diff
@@ -42,376 +39,347 @@
+    public void saveItem(String lemma, String item, String pos, String desc, String collectionId, String username)
+            throws QuadrigaStorageException, QuadrigaAccessException {
+        String errMsg = null;
+        try {
+            Query query = sessionFactory.getCurrentSession().createQuery(
+                    "from QuadrigaUserDTO user where user.username =:username and ( user.username IN (Select quadrigaUserDTO.username from ConceptCollectionCollaboratorDTO ccCollab where ccCollab.conceptCollectionDTO.conceptCollectionid =:conceptCollectionid) OR user.username IN (Select conceptColl.collectionowner.username from ConceptCollectionDTO conceptColl where conceptColl.conceptCollectionid =:conceptCollectionid))");
+            query.setParameter("username", username);
+            query.setParameter("conceptCollectionid", collectionId);
+
+            ConceptCollectionDTO conceptCollection = (ConceptCollectionDTO) sessionFactory.getCurrentSession()
+                    .get(ConceptCollectionDTO.class, collectionId);
+
+            if (conceptCollection == null) {
+                throw new QuadrigaStorageException();
+            }
+            List<QuadrigaUserDTO> quadrigaUserDTOList = query.list();
+            if (quadrigaUserDTOList != null && quadrigaUserDTOList.size() > 0) {
+                ConceptsDTO conceptsDTO = (ConceptsDTO) sessionFactory.getCurrentSession().get(ConceptsDTO.class, item);
+                if (conceptsDTO == null) {
+                    // create a new concept
+                    conceptsDTO = new ConceptsDTO();
+                    conceptsDTO.setItem(item);
+                    conceptsDTO.setCreateddate(new Date());
+                    conceptsDTO.setDescription(desc);
+                    conceptsDTO.setLemma(lemma);
+                    conceptsDTO.setPos(pos);
+                    conceptsDTO.setUpdateddate(new Date());
+                    sessionFactory.getCurrentSession().save(conceptsDTO);
+                }
+
+                // map the concept to the concept collection
+                ConceptCollectionItemsDTO conceptCollectionMapping = new ConceptCollectionItemsDTO();
+                ConceptCollectionItemsDTOPK conceptCollectionItemsDTOPK = new ConceptCollectionItemsDTOPK();
+                conceptCollectionItemsDTOPK.setConcept(item);
+                conceptCollectionItemsDTOPK.setConceptcollectionid(collectionId);
+                conceptCollectionMapping.setConceptcollectionsItemsDTOPK(conceptCollectionItemsDTOPK);
+                conceptCollectionMapping.setCreateddate(new Date());
+                conceptCollectionMapping.setUpdateddate(new Date());
+                sessionFactory.getCurrentSession().save(conceptCollectionMapping);
+
+            } else {
+                errMsg = "User dont have access to the collection";
+                logger.error(errMsg);
+                throw new QuadrigaAccessException("Hmmm!!  Need to try much more hard to get into this");

```

Comment:
remove the message that's passed to the exception (it's incorrect English and really doesn't mean anything). Just say what is actually happening (that the user doesn't have access to this collection).